seed/seed.go

```diff
@@ -116,6 +116,13 @@ type Seed interface {
 	// ModeSnaps returns the snaps that should be available
 	// in the given mode as defined by the seed, after LoadMeta.
 	ModeSnaps(mode string) ([]*Snap, error)
+
+	// NumSnaps returns the total number of snaps in a seed
+	NumSnaps() int
+
+	// Iter provides a way to iterately perform a function on
+	// each of the snaps in a seed
```

Comment:
same.   This needs to explain that for UC20 all snap are considered independent of mode. That's the reason I didn't rush to implement this originally. I didn't have use case. It's unclear whether f should take modes too, or Iter should take modes to filter on. It's probably ok as is for now but we need to document the behavior. What does your use case need?

Reply:
I'm going through and making these suggested changes now, but a specification of our use case would be better left here than in a comment. Our use case only needs to iterate through a list of pre-seeded snaps to gather their names and the channels that they were installed from.